trl/models/modeling_base.py

```diff
@@ -82,10 +88,31 @@ class and the arguments that are specific to trl models.
                 "pretrained_model_name_or_path should be a string or a PreTrainedModel, "
                 f"but is {type(pretrained_model_name_or_path)}"
             )
-
         # Then, create the full model by instantiating the wrapper class
         model = cls(pretrained_model, **trl_model_args)
 
+        # if resume_training, load the state_dict again - this is ok since the
+        # state_dict is removed from the model after loading it.
+        # since this is only done in edge cases we force the user to do it only
+        # if they explicitly set resume_training to True
+        if resume_training:
```

Comment:
If someone is using the `AutoModelForCausalLMWithValueHead` to load a model I think we can assume that they want to use the Value head so I we should always load it if it's available. What do you think?

Reply:
OK for me! 
We should advice users to use `AutoModelForCausalLM` when using the model outside `trl` (i.e. not for training)

Reply:
Added more description in [853d52d](https://github.com/lvwerra/trl/pull/86/commits/853d52d0e03f891c074a51c081da2a4e75dea40b)